### DISCUSSION TITLE
Reference previous row(s) in expressions

### DISCUSSION THREAD

**Discussion by madeloop** [perspective-dev/perspective#2041] [Q&A] [answered]
## Support Question

I'm not seeing any way to reference data from previous row(s) in expression columns. Is it correct that this is not possible?

Edit:
To add some context, I'm creating a table with financial trade data. If I would want to create a basic moving average, for example, it looks like I would have to either preprocess the data before it gets to perspective or maybe create a data flow with multiple perspective tables. Not sure about that second idea.

Update:
Found a discussion where it's mentioned that there is no way to reference other rows but it's a feature that was being worked on a year ago.

https://github.com/finos/perspective/discussions/1608

Is this on the roadmap?

**Comment by texodus** [comment, marked as answer]
Yes this core support is ready, but there are many functions that will "reference other rows" so which one you want depends on the answer.  I presume you are asking for e.g. `vlookup`, which in perspective will be by `index`?  This will be the first out.

**Reply by madeloop** [reply]
vlookup should allow me to do everything that I need for now if I'm able to access the previous row - something like "Colum Name"[1]

Are you able to provide a rough ETA for when you think this might be released?  Would love to work with vlookup but I have to get to building stuff this week so I'm not expecting it to be available that soon.

**Comment by madeloop** [comment]
Is the release of index(), window(), etc, still planned?

**Comment by cakir-enes** [comment]
Any updates on this?

**Comment by texodus** [comment]
#2374
